File: bench/py/33_sort_ints.py
# sort a list of ints (same LCG-generated data as the MyLang version)
import sys
scale = int(sys.argv[1]) if len(sys.argv) > 1 else 1

SZ = 200000
R = 3 * scale
checksum = 0

for k in range(R):

    a = [0] * SZ
    x = 123456789

    for i in range(SZ):
        x = (x * 1103515245 + 12345) % 2147483647
        a[i] = x

    a.sort()
    checksum += a[0] + a[SZ - 1]

print("result:", checksum)

File: bench/cpp/33_sort_ints.cpp
/* SPDX-License-Identifier: BSD-2-Clause */
/* Faithful C++ of bench/my/33_sort_ints.my: fill with an LCG, std::sort,
 * checksum the min+max. Same LCG + data as the .my/.py so the result agrees. */
#include "bench.h"
#include <algorithm>

int main(int argc, char **argv)
{
    long scale = bench_scale(argc, argv);
    const long SZ = 200000;
    long R = 3 * scale;
    long checksum = 0;

    for (long k = 0; k < R; k++) {
        std::vector<long> a(SZ, 0);
        long x = 123456789;
        for (long i = 0; i < SZ; i++) {
            x = (x * 1103515245 + 12345) % 2147483647;
            a[i] = x;
        }
        std::sort(a.begin(), a.end());
        checksum += a[0] + a[SZ - 1];
    }
    printf("result: %ld\n", checksum);
    return 0;
}
